target/linux/ar71xx/image/generic.mk

```diff
@@ -229,6 +229,84 @@ define Device/dragino2
 endef
 TARGET_DEVICES += dragino2
 
+define Device/e1700ac-8M
+  DEVICE_TITLE := WHQX E1700AC-8M (8MB flash)
+  DEVICE_PACKAGES := kmod-usb-core kmod-usb2 kmod-usb-ledtrig-usbport kmod-ath10k ath10k-firmware-qca988x
+  BOARDNAME := E1700AC
+  IMAGE_SIZE := 7744k
+  KERNEL_SIZE := 1472k
+  ROOTFS_SIZE := 6336k
+  MTDPARTS := spi0.0:256k(u-boot)ro,64k(u-boot-env)ro,6336k(rootfs),1472k(kernel),64k(art)ro,7744k@0x50000(firmware)
+  IMAGE/sysupgrade.bin := append-rootfs | pad-rootfs | \
+	pad-to $$$$(ROOTFS_SIZE) | append-kernel | check-size $$$$(IMAGE_SIZE)
+endef
+TARGET_DEVICES += e1700ac-8M
+
+define Device/e1700ac-16M
+  DEVICE_TITLE := WHQX E1700AC-16M (16MB flash)
+  DEVICE_PACKAGES := kmod-usb-core kmod-usb2 kmod-usb-ledtrig-usbport kmod-ath10k ath10k-firmware-qca988x
+  BOARDNAME := E1700AC
+  IMAGE_SIZE := 16000k
+  KERNEL_SIZE := 1472k
+  ROOTFS_SIZE := 14528k
+  MTDPARTS := spi0.0:256k(u-boot)ro,64k(u-boot-env)ro,14528k(rootfs),1472k(kernel),64k(art)ro,16000k@0x50000(firmware)
+  IMAGE/sysupgrade.bin := append-rootfs | pad-rootfs | \
+	pad-to $$$$(ROOTFS_SIZE) | append-kernel | check-size $$$$(IMAGE_SIZE)
```

Comment:
To add medata to the image, add append-metadata :
`$$(ROOTFS_SIZE) | append-kernel | append-metadata | check-size $$$$(IMAGE_SIZE)`
 and check if this image is accepted by "OEM" firmware.